pennylane/operation.py

```diff
@@ -1134,28 +1140,37 @@ def single_qubit_rot_angles(self):
         """
         raise NotImplementedError
 
-    def get_parameter_shift(self, idx, shift=np.pi / 2):
-        """Multiplier and shift for the parameter at position idx, based on its gradient recipe.
+    def get_parameter_shift(self, idx, shift=None):
+        r"""Multiplier and shift for the given parameter, based on its gradient recipe.
 
         Args:
             idx (int): parameter index
 
         Returns:
             list[[float, float, float]]: list of multiplier, coefficient, shift for each term in the gradient recipe
+
+        Note that the default value for ``shift`` is None, which is replaced by the
+        default shift :math:`\pi/2`.
         """
         # get the gradient recipe for this parameter
         recipe = self.grad_recipe[idx]
+        if recipe is not None:
+            return recipe
 
         # Default values
+        if shift is None:
+            shift = np.pi / 2
         multiplier = 0.5 / np.sin(shift)
         a = 1
 
         # We set the following default recipe:
         # ∂f(x) = c*f(a*x+s) - c*f(a*x-s)
         # where we express a positive and a negative shift by default
-        default_param_shift = [[multiplier, a, shift], [-multiplier, a, -shift]]
-        param_shift = default_param_shift if recipe is None else recipe
-        return param_shift
+        default_param_shift = [
+            [multiplier, a, shift],
+            [-multiplier, a, -shift],  # pylint: disable=invalid-unary-operand-type
```

Comment:
what an odd PyLint warning

Reply:
It is because `shift` defaults to `None` and `-None` would cause an error :D

Reply:
oh!